crates/bevy_pbr/src/light.rs

```diff
@@ -352,68 +487,63 @@ fn compute_aabb_for_cluster(
-        }
+    for (entity, config) in cameras.iter() {
+        let config = config.copied().unwrap_or_default();
+        // actual settings here don't matter - they will be overwritten in assign_lights_to_clusters
+        let clusters = Clusters::from_screen_size_and_dimensions(UVec2::ONE, UVec3::ONE, 1.0, 1.0);
+        commands.entity(entity).insert_bundle((clusters, config));
     }
 }
 
-pub fn update_clusters(
-    windows: Res<Windows>,
-    images: Res<Assets<Image>>,
-    mut views: Query<(&Camera, &mut Clusters)>,
+fn update_clusters(
+    screen_size: UVec2,
+    camera: &Camera,
+    cluster_dimensions: UVec3,
+    clusters: &mut Clusters,
+    near: f32,
+    far: f32,
 ) {
-    for (camera, mut clusters) in views.iter_mut() {
-        let is_orthographic = camera.projection_matrix.w_axis.w == 1.0;
-        let inverse_projection = camera.projection_matrix.inverse();
-        if let Some(screen_size_u32) = camera.target.get_physical_size(&windows, &images) {
-            // Don't update clusters if screen size is 0.
-            if screen_size_u32.x == 0 || screen_size_u32.y == 0 {
-                continue;
-            }
-            *clusters =
-                Clusters::from_screen_size_and_z_slices(screen_size_u32, clusters.axis_slices.z);
-            let screen_size = screen_size_u32.as_vec2();
-            let tile_size_u32 = clusters.tile_size;
-            let tile_size = tile_size_u32.as_vec2();
-
-            // Calculate view space AABBs
-            // NOTE: It is important that these are iterated in a specific order
-            // so that we can calculate the cluster index in the fragment shader!
-            // I (Rob Swain) choose to scan along rows of tiles in x,y, and for each tile then scan
-            // along z
-            let mut aabbs = Vec::with_capacity(
-                (clusters.axis_slices.y * clusters.axis_slices.x * clusters.axis_slices.z) as usize,
-            );
-            for y in 0..clusters.axis_slices.y {
-                for x in 0..clusters.axis_slices.x {
-                    for z in 0..clusters.axis_slices.z {
-                        aabbs.push(compute_aabb_for_cluster(
-                            clusters.near,
-                            camera.far,
-                            tile_size,
-                            screen_size,
-                            inverse_projection,
-                            is_orthographic,
-                            clusters.axis_slices,
-                            UVec3::new(x, y, z),
-                        ));
-                    }
-                }
+    let is_orthographic = camera.projection_matrix.w_axis.w == 1.0;
+    let inverse_projection = camera.projection_matrix.inverse();
+    // Don't update clusters if screen size is 0.
+    if screen_size.x == 0 || screen_size.y == 0 {
+        return;
+    }
+    *clusters =
+        Clusters::from_screen_size_and_dimensions(screen_size, cluster_dimensions, near, far);
```

Comment:
This panics for me in the `lighting.rs` example if i scale the window down to its minimimum size, then slowly scale it back up.
```
assertion failed: self.axis_slices.x * self.axis_slices.y * self.axis_slices.z <= 4096
```
The value of axis_slices on panic is `[16, 11, 24]`.

This is also true on main, but panicing for an internal renderer optimization isn't good, especially when a scene only has a few lights in it :)

Reply:
i have a fix - it's just a couple of lines. we discussed earlier whether to include it in this pr or a separate one. happy to plug it in here if you say so, otherwise i'll queue it up straight after this one.

Reply:
Awesome! It makes sense to handle it separately then. Thanks!